Tools/AP_Periph/AP_Periph.cpp

```diff
@@ -167,6 +167,14 @@ void AP_Periph_FW::init()
     hwesc_telem.init(hal.serial(3));
 #endif
 
+#ifdef HAL_PERIPH_ENABLE_ESC_APD
+    apd_esc_telem[0].init(hal.serial(APD_ESC_SERIAL_0), 32);
```

Comment:
please don't hardcode the serial port. See adsb_port and rangefinder_port as an example

Reply:
Tagging this as a question to the dev call (mostly seeking input from @tridge ). This is the model that's used in the HWESC telemetry, and it has some significant advantages that we don't have to check for nullptr's and the configuration is guaranteed to work. For flash reason's you will always have to opt in via the HWDEF, so specing the serial ports at the same time actually seems to make sense to me.

Reply:
we should have a HAL_PERIPH_APD_ESC_PORT_DEFAULT and use -1 to disable the feature. That way we can heave this feature builtin to some key builds, in particular f303-universal and CubeOrange-periph. That gives an easy to way to test without the specialist hardware and ensures it gets built in CI

Reply:
we also need the parameter to go along this with